homeassistant/components/watergate/entity.py

```diff
@@ -0,0 +1,50 @@
+"""Watergate Base Entity Definition."""
+
+from dataclasses import dataclass
+
+from watergate_local_api import WatergateLocalApiClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER
+from .coordinator import WatergateDataCoordinator
+
+type WatergateConfigEntry = ConfigEntry[WatergateData]
+
+
+class WatergateEntity(CoordinatorEntity[WatergateDataCoordinator]):
+    """Define a base Watergate entity."""
+
+    def __init__(
+        self,
+        coordinator: WatergateDataCoordinator,
+        entry: WatergateConfigEntry,
+        entity_name: str,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator)
+        self._attr_has_entity_name = True
+        self._api_client = coordinator.api
+        self._attr_unique_id = f"{entry.data[CONF_NAME]}.{entity_name}"
+        self._entry = entry
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, entry.entry_id)},
+            name=entry.data[CONF_NAME],
+            manufacturer=MANUFACTURER,
+            sw_version=coordinator.data.state.firmware_version
+            if coordinator.data.state
+            else "",
+        )
+        self._api_client = entry.runtime_data.client
+
+
+@dataclass
+class WatergateData:
+    """Data for the Watergate integration."""
+
+    client: WatergateLocalApiClient
+    coordinator: WatergateDataCoordinator
+    sonic_name: str
```

Comment:
Why does this belong here?

Reply:
It is required to create the entity's unique name, so it will be removed soon since I've already asked our engineers to introduce the Serial number in the device state response and I will use that plus the entity name.

Reply:
No I mean, why does this class belong in this file

Reply:
If I remember correctly I had some circular dependencies and I thought that this is the best place since it is related to Entity. Where by default do you store those?

Reply:
It's not related to the entity, rather to the integration on runtime, so I would propose coordinator.py or init.py. But in theory you don't even need this class anymore